modules/navier_stokes/src/postprocessors/NSCFLTimeStepSize.C

```diff
@@ -0,0 +1,108 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// Navier-Stokes includes
+#include "NSCFLTimeStepSize.h"
+
+// MOOSE includes
+#include "libmesh/quadrature.h"
+#include "metaphysicl/raw_type.h"
+
+#include <algorithm>
+#include <limits>
+
+registerMooseObject("NavierStokesApp", CFLTimeStepSize);
+registerMooseObject("NavierStokesApp", ADCFLTimeStepSize);
```

Comment:
why do we need an AD version of this?

Reply:
Because this retrieves material properties and has to retrieve either non-AD or AD properties

Reply:
ugh this is true but also ugh. Is there a getNonADMaterialPropertyWhetherItsADorNot() routine?

Reply:
No. We are using a statically typed language so there is no way around this. We used to allow inter-conversion between non-AD and AD material properties but it was disastrous